docs/writing_review_screen.md

```diff
@@ -0,0 +1,240 @@
+    ```
+
+    Most notably is that there is a `review` attribute instead of a `fields`. Each element under the `review` attribute
+    is a different section, generated from screens in the interview, with the subquestion and values of the fields shown.
+
+    Each element in the review can have the following dictionary attributes:
+
+    * `Edit` adds an "Edit" button next to the information on the review screen. It's value is the name of the variable that will be edited when the button is clicked.
+    * `label` lets you change the text on the "Edit" button
+    * `fields` lets you add multiple fields that are re-asked to the user for a section on the review screen.
+      * `recompute` lets you also re-calculate other variables, not necessarily re-asking them, but letting them be filled by `code` blocks. These won't be generated, but will come in handy later.
+    * `button` is the actual content of that section of the review screen, similar to a `subquestion` attribute.
+
+3. The revisit screens. These blocks look like this:
+
+    ```yml
+    id: revisit users
+    continue button field: users.revisit
+    question: |
+      Edit your answers about Users
+    subquestion: |
+      ${ users.table }
+
+      ${ users.add_action() }
+    ---
+    ...
+    ```
+
+    Each block has the table and lets people edit items in the lists.
+
+4. The generated tables for the lists.
+
+    ```yml
+    table: children.table
+    rows: children
+    columns:
+      - Birthdate: |
+          row_item.birthdate if hasattr(row_item, 'birthdate') else ''
+    edit:
+      - birthdate
+    ```
+
+### Step 3: Edits to the Review Screen Generator
+
+You'll want to change some aspects of the generated review screen. Note that if any errors happen in a specific edit element on the review screen, that element won't appear. It also won't appear if the attribute or variable to be edited doesn't exist. You should make sure that it does.
+
+Some common issues to the generated review screen that I personally have run into before:
+
+1. Add tables for people lists, like `users` or `other_parties`. The below block is a good start.
+
+```yml
+generic object: ALPeopleList
+table: x.table
+rows: x
+columns:
+  - Name: |
+      row_item.name if defined('row_item.name.first') else ''
+  - Address: |
+      row_item.address if defined('row_item.address.address') else ''
+  - Phone number: |
+      row_item.phone_numbers()
+edit:
+  - name.first
+  - address.address
+  - phone_number
+delete buttons: False
+```
+
+2. Remove tables for attachments and other non-people objects, if any were generated. Tables are only needed for information that the user enters themselves, which the review screen generator can't determine yet.
+3. Remove edit questions about kickout screens. Sometimes, you will have questions that do set a variable in the interview, but it's only use is to kickout a user if their problem isn't covered by the form. Here's an example of one:
+
```

Comment:
I'd make the heading "Remove questions you don't want the user to edit" and make the kickout screen an example.